### ISSUE TITLE
Passing variable to @classmethod super() triggers invalid "[Pyright reportGeneralTypeIssues] [E]"

### ISSUE THREAD

**Issue by Kache** [microsoft/pyright#1875]
**Describe the bug**

Incorrectly getting:
`[Pyright reportGeneralTypeIssues] [E] Expected class type as first argument to "super" call but received "Type[Bar]"`

**To Reproduce**

```python
class Foo:
    def __init__(self):
        self.x = 'foo'


class Bar(Foo):
    @classmethod
    def from_str(cls, s):
        obj = cls.__new__(cls)
        super(cls, obj).__init__()  # on this line, over `cls`
        # super(Bar, obj).__init__()  # doesn't error, but this line is equivalent
        obj.x += 'bar' + s
        return obj


print(Bar.from_str('hello').x)
```

**Expected behavior**

As I understand it, `from_str` is a [standard way of creating an "alternative" initializer](https://stackoverflow.com/a/38885481/234593) in Python, so there should be no such error.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
def foo(self) -> str:
    return 3
```

**VS Code extension or command-line**

[Both VS Code and CLI reproduction screenshots by `coc-pyright` maintainer](https://github.com/fannheyward/coc-pyright/issues/484#issuecomment-842854073)

I can download VS Code and get version information if needed.

**Additional context**

Moved upstream from [coc-pyright](https://github.com/fannheyward/coc-pyright/issues/484)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of pyright.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.141, which I just published. It will also be included in the next release of pylance.
